azure-pipelines.yml

```diff
@@ -136,6 +153,22 @@ jobs:
       publishVstsFeed: '2cac8390-d7aa-4320-b481-1a32c53de85b'
       allowPackageConflicts: true
 
+  - task: PowerShell@1
+    condition: eq(variables['Build.SourceBranch'], 'refs/heads/master')
+    displayName: 'Push VSIX'
+    inputs:
+      scriptType: inlineScript
+      inlineScript: |
+        # Setting these environment variables because they are used in the script for the upload endpoint: 
+     
+        $env:APPVEYOR_REPO_PROVIDER = "github"
+        $env:APPVEYOR_REPO_NAME = "microsoft/dotnet-apiport"
+     
+        "tools\AppVeyor\vsix.ps1" | Invoke-Expression
+        $vsix = gci -path bin\$(BuildConfiguration) -Recurse -Include ApiPort.vsix -File
+        $vsix
+        Vsix-PublishToGallery -path $vsix 
```

Comment:
I would not publish to an external gallery directly from my build pipeline. Instead, I recommend you define a second YAML file that consumes the output of this one. See [this example](https://github.com/AArnott/Library.Template/blob/microbuild/azure-pipelines/release.yml) of a release pipeline that consumes the build pipeline.
You can even create a trigger so that whenever you push to master the release pipeline is triggered, thereby removing extra conditions and simplifying your regular build pipeline.

Reply:
@AArnott Thanks for your feedbacks! Submitted a new commit with
1. Created a separated release pipleline.
2. Moved common code for both pipeline into two *common*.yml files and used them as template files.